brave/brave-core
# Queue ad events until ads library is initialized.

Resolves https://github.com/brave/brave-browser/issues/31524



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1) Check that events are triggered for Brave Ads. Example of events:
    * Browser entered the foreground
    * Browser entered the background
    * Tab text content did change
    * Tab HTML content did change
    * Tab did start/stop media
    * Tab closed
    * Preference changed

2) Check that events are not triggered until Brave Ads are initialized
